### PR TITLE
resnet model

### PR DESCRIPTION
**Pull request by TonioBall**
added eegresnet.py with residual network model

### CODE REVIEW

**Comment by robintibor** [comment]
Great

File: braindecode/models/eegresnet.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+from torch import nn
+from torch.nn import init
+import torch as th
+from torch.nn.functional import elu
+
+from braindecode.models.base import BaseModel
+
+from braindecode.models.modules import Expression, AvgPool2dWithConv
+from braindecode.models.functions import identity
+from braindecode.models.modules import np_to_var
```
Comment (agramfort) [inline braindecode/models/eegresnet.py]: you should use relative imports in the library

File: braindecode/models/eegresnet.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+        # Initialize all weights
+        model.apply(lambda module: weights_init(module, self.conv_weight_init_fn))
+
+        # Start in eval mode
+        model.eval()
+        return model
+
+
+def weights_init(module, conv_weight_init_fn):
```
Comment (agramfort) [inline braindecode/models/eegresnet.py]: I would make this function private starting with _

File: braindecode/models/eegresnet.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        x = x[:,:,0]
+    return x
+
+
+def _transpose_time_to_spat(x):
+    return x.permute(0, 3, 2, 1)
+
+
+# create a residual learning building block with two stacked 3x3 convlayers as in paper
+class ResidualBlock(nn.Module):
```
Comment (agramfort) [inline braindecode/models/eegresnet.py]: I assume it's not a public class so I would call it _ResidualBlock

File: braindecode/models/eegresnet.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+from braindecode.models.base import BaseModel
+
+from braindecode.models.modules import Expression, AvgPool2dWithConv
+from braindecode.models.functions import identity
+from braindecode.models.modules import np_to_var
+
+
+class EEGResNet(BaseModel):
+    """
+    Residual Network for EEG.
```
Comment (agramfort) [inline braindecode/models/eegresnet.py]: for new code let's force ourself to add docstrings

Comment (agramfort) [reply braindecode/models/eegresnet.py]: it should also have been added to the api.rst page

Comment (TonioBall) [reply braindecode/models/eegresnet.py]: hope docstrings are ok/better now?

**Pull request merged**